WordPress/openverse-catalog
# Science Museum: Handle unrecognized licenses

## Fixes


Fixes #738 by @AetherUnbound

## Description


This PR addresses an issue wherein licenses which were expected to follow the CC license pattern actually ended up being generic license types. I have changed the tests to use in-line data, and added all of the cases I was able to capture that caused issues when I ran this locally.

I also added a new year range, 2020-2023, so we can include data from the last two years :smile:

## Testing Instructions

`just test`

If you want to observe the failure:

1. Check out `main`
2. Comment out all year ranges except `(0, 1500)` (this was the one that originally caused the error)
3. **Disable your ingestion limit**
4. Run the DAG, and note that the error in the issue occurs
5. Check out this branch
6. Ensure that the othe year ranges are still commented out
7. Run the DAG again, observe that it completes!

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
